UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -0,0 +1,370 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Health.Sickness;
+using HealthV2;
+using JetBrains.Annotations;
+using Mirror;
+using UnityEngine;
+
+public abstract class LivingHealthMasterBase : NetworkBehaviour
+{
+	/// <summary>
+	/// Server side, each mob has a different one and never it never changes
+	/// </summary>
+	public int mobID { get; private set; }
+
+	private DNAandBloodType DNABloodType;
+
+	// JSON string for blood types and DNA.
+	[SyncVar(hook = nameof(DNASync))] //May remove this in the future and only provide DNA info on request
+	private string DNABloodTypeJSON;
+
+	private float tickRate = 1f;
+	private float tick = 0;
+
+	private RegisterTile registerTile;
+	public RegisterTile RegisterTile => registerTile;
+
+	[NonSerialized] public ConsciousStateEvent OnConsciousStateChangeServer = new ConsciousStateEvent();
+	public ConsciousState ConsciousState
+	{
+		get => consciousState;
+		protected set
+		{
+			ConsciousState oldState = consciousState;
+			if (value != oldState)
+			{
+				consciousState = value;
+				if (isServer)
+				{
+					OnConsciousStateChangeServer.Invoke(oldState, value);
+				}
+			}
+		}
+	}
+	private ConsciousState consciousState;
+
+	//I don't know what this is. It was in the first Health system.
+	public float RTT;
+
+	[SerializeField]
+	private float maxHealth = 100;
+
+	private float overallHealth = 100;
+	public float OverallHealth => overallHealth;
+
+	[SerializeField]
+	[Tooltip("These are the things that will hold all our organs and implants.")]
+	private List<BodyPartContainerBase> bodyPartContainers;
+
+	[CanBeNull]
+	private CirculatorySystemBase circulatorySystem;
+	public CirculatorySystemBase CirculatorySystem => circulatorySystem;
+
+	[CanBeNull]
+	private RespiratorySystemBase respiratorySystem;
+	public RespiratorySystemBase RespiratorySystem => respiratorySystem;
+
+	[CanBeNull]
+	private MetabolismSystemV2 metabolism;
+	public MetabolismSystemV2 Metabolism => metabolism;
+
+	private bool isDead = false;
+	public bool IsDead => isDead;
+
+	public bool IsCrit => ConsciousState == ConsciousState.UNCONSCIOUS;
+	public bool IsSoftCrit => ConsciousState == ConsciousState.BARELY_CONSCIOUS;
+
+	public bool InCardiacArrest => circulatorySystem != null && circulatorySystem.HeartIsStopped;
+
+	private List<ImplantBase> implantList = new List<ImplantBase>();
+
+	public List<ImplantBase> ImplantList => implantList;
+
+	public event Action<GameObject> applyDamageEvent;
```

Comment:
Codacy found an issue: [Remove the unused event 'applyDamageEvent' or invoke it.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6324510)

Reply:
This is required